python/lsst/sconsUtils/tests.py

```diff
@@ -235,13 +240,16 @@ def run(self, fileGlob):
         # to the command.
         libpathstr = utils.libraryLoaderEnvironment()
 
+        # If user has set an envvar to control warnings, defer to them.
+        warn_opt = "-Wd " if "PYTHONWARNINGS" not in os.environ else ""
```

Comment:
I think that `PYTHONWARNINGS` extends (and potentially overrides) anything specified with `-W`, according to my reading of https://docs.python.org/3/library/warnings.html#describing-warning-filters.  So I'm not sure that it's necessary to remove `-Wd`.

Reply:
Good catch; fixed.  I was led astray by previous versions of this change that suffered from other problems.  I've tested this with my ticket branches for DM-31924 and I can confirm that I see test failures for deprecated interface usage with PYTHONWARNINGS set and these changes removed.